src/address.rs

```diff
@@ -44,6 +44,17 @@ pub struct Address<A, Rc: RefCounter = Strong> {
     pub(crate) ref_counter: Rc,
 }
 
+impl<A, Rc: RefCounter> Debug for Address<A, Rc> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let a = std::any::type_name::<A>();
+        let rc = std::any::type_name::<Rc>();
+
+        f.debug_struct(&format!("Address<{}, {}>", a, rc))
+            .field("ref_counter", &self.ref_counter)
```

Comment:
The strong/weak part is duplicated here, isn't it? I think we can remove it from the `Address` type itself and only have it as part of the `ref_counter` field?

Reply:
That is technically true. I mostly kept it to show the full type since  Address<A> might seem to imply strong as that's the default

Reply:
Fair enough!
Let's see what it looks like when printed but it would be nice to reduce the duplication one way or another :)